src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/ALoadoutSynchronizer.cs

```diff
@@ -457,93 +519,111 @@ private void ActionDeleteFromDisk(SyncActionGroupings<SyncTreeNode> groupings, I
         {
             var gamePath = register.GetResolvedPath(item.Path);
             gamePath.Delete();
-            previousTree.Remove(item.Path);
+
+            // Don't delete the entry if we're just going to replace it
+            if (!item.Actions.HasFlag(Actions.ExtractToDisk))
+            {
+                var id = item.Disk.Value.Id;
+                tx.Retract(id, DiskStateEntry.Path, item.Disk.Value.Path);
+                tx.Retract(id, DiskStateEntry.Hash, item.Disk.Value.Hash);
+                tx.Retract(id, DiskStateEntry.Size, item.Disk.Value.Size);
+                tx.Retract(id, DiskStateEntry.LastModified, item.Disk.Value.LastModified);
+                tx.Retract(id, DiskStateEntry.Game, item.Disk.Value.Game);
+            }
         }
     }
 
-    private async Task ActionBackupFiles(SyncActionGroupings<SyncTreeNode> groupings, Loadout.ReadOnly loadout)
+    private async Task ActionBackupFiles(SyncActionGroupings<SyncTreeNode> groupings, GameInstallation gameInstallation)
     {
         var toBackup = groupings[Actions.BackupFile];
         _logger.LogDebug("Backing up {Count} files", toBackup.Count);
-                    
-        await BackupNewFiles(loadout.InstallationInstance, toBackup.Select(item =>
-            (item.Path, item.Disk.Value.Hash, item.Disk.Value.Size)));
+
+        await BackupNewFiles(gameInstallation, toBackup.Select(item =>
+                (item.Path, item.Disk.Value.Hash, item.Disk.Value.Size)
+            )
+        );
+    }
+
+    public record struct AddedEntry
+    {
+        public required LoadoutItem.New LoadoutItem { get; init; }
+        public required LoadoutItemWithTargetPath.New LoadoutItemWithTargetPath { get; init; }
+        public required LoadoutFile.New LoadoutFileEntry { get; init; }
     }
```

Comment:
We don't need to store the `.New` variants of the parents, this can just have `LoadoutFile.New`:

```csharp
    public record struct AddedEntry
    {
        public required LoadoutFile.New LoadoutFile { get; init; }
        public required ITransaction Transaction { get; init; }

        public LoadoutItemWithTargetPath.New LoadoutItemWithTargetPath => LoadoutFile.GetLoadoutItemWithTargetPath(Transaction);
        public LoadoutItem.New LoadoutItem => LoadoutItemWithTargetPath.GetLoadoutItem(Transaction);
    }
```

I don't think we need this extra record type at all, since you can get to everything from just passing `LoadoutFile.New`.

Reply:
Those objects don't store pointers to eachother, however, so if we don't have this object we have to look each variant of `.New` up in the ITransaction every time we need it. I'd prefer to pass this information in via the method call instead of having every implementation have to perform a `O(n)` scan of all the items in the transaction to find the matching objects.

Reply:
Probably fine for now, but this will likely be a recurring issue. We might want to rethink how we get "parent" `.New` values.